http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/IamProxyAuthProvider.scala

```diff
@@ -168,4 +182,35 @@ class IamProxyAuthProvider(config: Config, saProvider: ServiceAccountProvider[Fu
     runtimeName: RuntimeName
   )(implicit ev: ApplicativeAsk[Future, TraceId]): Future[Unit] = Future.unit
 
+  /**
+    * Leo calls this method to notify the auth provider that a new persistent disk has been created.
+    * The returned future should complete once the provider has finished doing any associated work.
+    * Leo will wait, so be timely!
+    *
+    * @param internalId     The internal ID for the disk (i.e. used for Sam resources)
+    * @param creatorEmail   The email address of the user in question
+    * @param googleProject  The Google project the disk was created in
```

Comment:
Note: For the functions that deal with runtimes, a RuntimeName is being passed around but not used, so I elected not to pass around a DiskName for the ones that deal with PDs